test/foundry/dapp/Application.t.sol

```diff
@@ -64,6 +76,8 @@ contract ApplicationTest is ERC165Test {
     uint256 constant _initialSupply = 1000000000000000000000000000000000000;
     uint256 constant _tokenId = 88888888;
     uint256 constant _transferAmount = 42;
+    uint256[] _ERC1155IdArray = _generateUint256Array(7);
+    uint256[] _ERC1155SupplyArray = _generateUint256Array(7);
```

Comment:
We're missing an array of transfer amounts so that we can actually test the balance after the transfer being non-zero. Also, I think we can come up with better names for these. Why not `_initialSupplies`, `_tokenIds`, and `_transferAmounts`?

Also, I'd be fine with having dynamic state variables only be initialized in `_initVariables`, just like `_interfaceIds`.  We could just loop over the number of token types, and push elements to these arrays one-by-one.

In my opinion, IDs can remain sequential (but then, `_generateUint256Array` wouldn't be necessary, as we'd have the index `i` readily available in the loop anyway, so we could remove that function), total supplies can be constant (and even equal to `_totalSupply`, a sensible default), and transfer amounts should be random and bounded by total supply.